### ISSUE TITLE
Collect SMART data with metricbeat

### ISSUE THREAD

**Issue by jsoriano** [elastic/beats#8614]
Data provided by [SMART](https://en.wikipedia.org/wiki/S.M.A.R.T.) is useful to detect failures on disks or to prevent incidences caused by disk degradation. It'd be nice to add a system metricset that collects this information.

SMART data is usually collected using [smartmontools](https://www.smartmontools.org/), we may need to find a golang-based alternative for that.

**Comment by sayden** [comment]
I found this library that could potentially help with the implementation https://github.com/dswarbrick/smart

**Comment by jsoriano** [comment]
Other option could be to parse `smartd` logs with filebeat.

**Comment by agathver** [comment]
I would like to work on this

**Comment by jsoriano** [comment]
@agathver great! I don't think anyone is working on this yet, so you are welcome to start with this :slightly_smiling_face:

What is the approach you want to try?

Thanks!

**Comment by agathver** [comment]
@jsoriano I feel the smartmontools approach is more robust and mature. The Go library will be easier but has a warning of being experimental.

We already have other modules which require additional software, so this should not be an issue IMO.

**Comment by jsoriano** [comment]
I agree in any case that using smartmontools if possible would be preferable to other options.

**Comment by agathver** [comment]
My initial idea was to parse the output of `smartctl`. The other alternative is to parse logs by `smartd` which goes into syslog.

**Comment by cskowronnek** [comment]
Any news here? Is there a way to pull SMART data from Windows devices?

**Comment by fearful-symmetry** [comment]
This sounds fun to pick up. I need to look into how smartctl works, but if it's just doing ioctl calls to some disk device it probably won't be that hard to re-implement.

**Comment by agathver** [comment]
@fearful-symmetry You can take a look at WMI as well, it provides SMART data as well

**Comment by guillaumedsde** [comment]
kind bump :) I'd be quite interested in this feature

**Comment by suprematis** [comment]
Bump here too!!

**Comment by fearful-symmetry** [comment]
Gonna keep this open and see if we can at least keep it on our radar.

**Comment by PrplHaz4** [comment]
this sounds like it could be related to #20562

**Comment by HugoKuo** [comment]
bump! This will be very useful for AI modeling to detect potential failure disk.

**Comment by LMey** [comment]
Is there any chance that the WMI functionality from Logstash could be moved into metricbeats?

Reasons:
As I understand logstash is a central function. So using this requires remote calls and security configuration
metric beat is where metrics are collate and set through to elastic. So why go somewhere else for your metrics.

It is also requested here: https://github.com/elastic/beats/issues/29072

**Comment by Erikg346** [comment]
+1 Is there any update on this?
